xarray/core/combine.py

```diff
@@ -456,107 +146,372 @@ def _combine_nd(combined_ids, concat_dims, data_vars='all',
         keys, which specify position within the desired final combined result.
     concat_dims : sequence of str
         The dimensions along which the datasets should be concatenated. Must be
-        in order, and the length must match
+        in order, and the length must match the length of the tuples used as
+        keys in combined_ids. If the string is a dimension name then concat
+        along that dimension, if it is None then merge.
 
     Returns
     -------
     combined_ds : xarray.Dataset
     """
 
-    # Perform N-D dimensional concatenation
+    tile_id, ds = list(combined_ids.items())[0]
```

Comment:
A few minor suggestions:
- If you don't need a variable (e.g., `ds` in this case), then it's convention to assign it to the name `_`, which indicates that it isn't used. But in this case, I think it would work simply to use `.keys()` instead of `.items()`.
- Rather than constructing a list, we can get the first element just by use the builtins `iter()` and `next()`. In principle this could be much faster, because it avoids constructing a list just to look at only the first item.
- Maybe use a more descriptive name than `tile_id`?

So this could be:
```suggestion
    example_tile_id = next(iter(combined_ids.keys()))
```